lib/crypto.js

```diff
@@ -0,0 +1,166 @@
+}
+
+/**
+ * Takes a Buffer or Uint8Array and adds a DER Private Prefix to it.
+ * Allows Uint8Arrays to be interoperable with node's crypto functions.
+ *
+ * @param {object} options - Options to use.
+ * @param {Buffer} [options.privateKeyBytes] - Required if no seedBytes.
+ * @param {Buffer} [options.seedBytes] - Required if no privateKeyBytes.
+ *
+ * @throws {TypeError} Throws if the supplied buffer is not of the right size
+ *  or not a Uint8Array or Buffer.
+ *
+ * @returns {Buffer} DER private key prefix + key bytes.
+*/
+function privateKeyDerEncode({privateKeyBytes, seedBytes}) {
+  if(!(privateKeyBytes || seedBytes)) {
+    throw new TypeError('`privateKeyBytes` or `seedBytes` is required.');
+  }
+  if(!privateKeyBytes && !(seedBytes instanceof Uint8Array &&
+    seedBytes.length === 32)) {
+    throw new TypeError('`seedBytes` must be a 32 byte Buffer.');
+  }
+  if(!seedBytes && !(privateKeyBytes instanceof Uint8Array &&
+    privateKeyBytes.length === 64)) {
+    throw new TypeError('`privateKeyBytes` must be a 64 byte Buffer.');
+  }
+  let p;
+  if(seedBytes) {
+    p = seedBytes;
+  } else {
+    // extract the first 32 bytes of the 64 byte private key representation
+    p = privateKeyBytes.slice(0, 32);
+  }
+  return Buffer.concat([DER_PRIVATE_KEY_PREFIX, p]);
+}
+
+/**
+ * Takes a Buffer of public key bytes and adds the DER Public Key Prefix to it.
+ * Allows Uint8Arrays to be interoperable with node's crypto functions.
+ *
+ * @param {object} options - Options to use.
+ * @param {Uint8Array} options.publicKeyBytes - The keyBytes.
+ *
+ * @throws {TypeError} Throws if the bytes are not Uint8Array or of length 32.
+ *
+ * @returns {Buffer} DER Public key Prefix + key bytes.
+*/
+function publicKeyDerEncode({publicKeyBytes}) {
+  if(!(publicKeyBytes instanceof Uint8Array && publicKeyBytes.length === 32)) {
+    throw new TypeError('`publicKeyBytes` must be a 32 byte Buffer.');
+  }
+  return Buffer.concat([DER_PUBLIC_KEY_PREFIX, publicKeyBytes]);
+}
+
+// FIXME someone more familar with TypedArrays and Buffers
+// should review this
+const typedArrayToBuffer = array => {
+  const length = array.byteLength + array.byteOffset;
+  const arrayBuffer = array.buffer.slice(array.byteOffset, length);
+  return Buffer.from(arrayBuffer, array.byteOffset, length);
+};
+
+// seedBytes can be a TypedArray or a function that returns a TypedArray
+const getSeedBytes = async seed => {
+  if(typeof seed === 'function') {
+    return seed();
+  }
+  return seed;
+};
+
+// helper function that takes an TypedArray of private bytes and turns it
+// into a Buffer containing a DER encoded key
+const privateBytesToDer = array => privateKeyDerEncode(
+  {privateKeyBytes: typedArrayToBuffer(array)});
+
+// helper function that takes a TypedArray of public bytes and turns it
+// into a Buffer containing a DER encoded key
+const publicBytesToDer = array => publicKeyDerEncode(
+  {publicKeyBytes: typedArrayToBuffer(array)});
```

Comment:
Can we remove these entirely and just call `privateKeyDerEncode` and `publicKeyDerEncode` directly where needed? We may want to use the names `privateKeyToDer` and `publicKeyToDer` instead though.

Reply:
Addressed here and good idea: https://github.com/digitalbazaar/ed25519-verification-key-2020/pull/3/commits/7efbbfd8d01fe357a5593ec4095533ea8d108c96